### PR TITLE
Adding support for locale string comparison for predicates #791.

### PR DESCRIPTION
**Pull request by msft-Jeyaram**
Moving some code to utlity.
Fixing unit tests on OSX #746

### CODE REVIEW

File: tests/unittests/Foundation/NSComparisonPredicateTests.m [hunk 1]
```diff
@@ -630,7 +630,7 @@ - (void)dealloc {
 
     ASSERT_TRUE_MSG(comparisonPredicate != nil, "FAILED: comparisonPredicate should be non-null!");
 
-    ASSERT_TRUE_MSG([comparisonPredicate evaluateWithObject:nil substitutionVariables:nil], "FAILED: Comparison failed.");
+    ASSERT_TRUE_MSG(![comparisonPredicate evaluateWithObject:nil substitutionVariables:nil], "FAILED: Comparison failed.");
```
Comment (rajsesh-msft) [inline tests/unittests/Foundation/NSComparisonPredicateTests.m]: >(

This is the exact opposite of before, what happened?

Comment (msft-Jeyaram) [inline tests/unittests/Foundation/NSComparisonPredicateTests.m]: this had to do with an issue with capitalization option.  we didn't support the full fledged options. now we do.

Comment (msft-Jeyaram) [inline tests/unittests/Foundation/NSComparisonPredicateTests.m]: are vs Are

---
In reply to: [73914570](https://github.com/Microsoft/WinObjC/pull/846#discussion_r73914570) [](ancestors = 73914570)

**Comment by msft-Jeyaram** [comment]
In CI Build

**Comment by msft-Jeyaram** [comment]
Ping!

**Comment by rajsesh-msft** [comment]
:shipit:

**Pull request merged**